src/components/ui/DataGrid/DataGrid.svelte

```diff
@@ -80,6 +80,23 @@
   let previousSelectedRowId: RowId | null = null;
 
   $: {
+    const seen_set = new Set<number>();
+    rowData.forEach(rowDatum => {
+      if (!seen_set.has(rowDatum[idKey])) {
+        // Non duplicate case
+        seen_set.add(rowDatum[idKey]);
+      } else {
+        // Found duplicate, write error message
+        console.error(
+          `%c Grid Problems? Look Here!
+A DataGrid has had multiple rows keyed over the same ID - ensure no two rows have the same value for the \`${String(
+            idKey,
+          )}\` property at the same time, even for a moment.
```

Comment:
Thoughts on logging out the offending duplicate key?

Reply:
Hmmm. It could maybe just be a small statement at the end saying something like: `The duplicate ${String(idKey)} value is:  ${rowDatum[idKey]}.`

Reply:
> Thoughts on logging out the offending duplicate key?

Sorry...I don't read...for some reason I thought @Cobular replied with this in response to my other comment so i was providing a suggestion. 🤦🏻 Please don't take this as a dictation that it should state the duplicate key value if you don't think that it should @Cobular.